### PR TITLE
samples: nrf9160: lwm2m_client: Make software version configurable

### PR DESCRIPTION
**Pull request by rlubos**
Currently Software Version resource of the Device object was set within
lwm2m_connmon.c, which is counter intuitve.

Move the resource intiialization to the Device object handler. Add
Kconfig config, which allow to override the default software version.
Use NCS version by default.

Signed-off-by: Robert Lubos <robert.lubos@nordicsemi.no>

### CODE REVIEW

File: subsys/net/lib/lwm2m_client_utils/lwm2m/lwm2m_connmon.c [hunk 1]
```diff
@@ -87,11 +87,6 @@ static void modem_data_update(struct k_work *work)
 		modem_param.device.modem_fw.value_string,
```
Comment (VeijoPesonen) [inline subsys/net/lib/lwm2m_client_utils/lwm2m/lwm2m_connmon.c:87]: How about moving 'Firmware Version' too?

**Review by VeijoPesonen — COMMENTED** [review]
Question about 'Firmware Version'. Otherwise looks good to me.

**Comment by rlubos** [comment]
Added new option description to the readme file.

File: samples/nrf9160/lwm2m_client/README.rst [hunk 2]
```diff
@@ -369,6 +369,9 @@ LwM2M objects options
 
    The configuration option sets the version of the OMA IPSO object specification that is to be used by the user defined Color sensor IPSO object to 1.1.
 
+.. option:: APP_CUSTOM_VERSION - Configuration to set custom application version reported in the Device object.
+
+   The configuration option allows to specify custom application version reported to the LwM2M server. By default, the current NCS version is used.
```
Comment (umapraseeda) [inline samples/nrf9160/lwm2m_client/README.rst:374]:
```suggestion
   The configuration option allows to specify custom application version reported to the LwM2M server. By default, the current |NCS| version is used.
```

**Pull request merged**